common/src/provisioning/Authentication/ServiceConnectionString.cs

```diff
@@ -19,10 +19,9 @@ namespace Microsoft.Azure.Devices.Common.Service.Auth
     /// 
     /// This object parse and store the connection string. It is responsible to provide the authorization token too. 
     /// </remarks>
-    internal sealed class ServiceConnectionString : IAuthorizationHeaderProvider
+    public sealed class ServiceConnectionString : IAuthorizationHeaderProvider
```

Comment:
I don't think it makes sense for this class to derive from the IAuthHeaderProvider.